### PR TITLE
Add f.mapFailure(new CustomException); Add option.toFuture (without throwable)

### PR DESCRIPTION
**Pull request by michalkowol**
```
val option = Option(7)
option.toFuture

val future = Future(7)
future.mapFailure(new Exception("foo"))
```

### CODE REVIEW

**Comment by michalkowol** [comment]
@2rs2ts: test are flapping on Travis (#108).
The same commit (54f01f6) on my github space passed all [tests](https://travis-ci.org/michalkowol/cascade/builds/54350489).

**Comment by 2rs2ts** [comment]
@michalkowol Yeah, we're going to have to look at that later this week. I'll have one of my coworkers co-review this PR but it looks good to me. Thanks for the contribution!

File: common/src/main/scala/com/paypal/cascade/common/option/option.scala [hunk 1]
```diff
@@ -92,6 +94,15 @@ package object option {
     def toFuture(t: => Throwable): Future[T] = {
       option.map(Future.successful).getOrElse(Future.failed(t))
     }
+
+    /**
+     * Map the Option into Future success/fail states: if the Option is a Some, yield a Future with its value;
+     * if the Option is None, yield a Future with NoSuchElementException as its failure state
+     * @return the Future-wrapped value of this Option, or a Future-wrapped failure Throwable
+     */
+    def toFuture: Future[T] = {
```
Comment (taylorleese) [inline common/src/main/scala/com/paypal/cascade/common/option/option.scala]: @michalkowol @2rs2ts I would actually expect this to return a ```Future[Option[T]]```. Getting a result of a failed future without passing in a function that creates a a throwable like on line 94 would be a bit of a surprise to me.

Comment (michalkowol) [inline common/src/main/scala/com/paypal/cascade/common/option/option.scala]: My intention was to have the most consistent code as possible.
Now we have:
```scala
"some".toRight[Throwable]
      .toTry
      .toOption
      .toFuture(new Exception("bar"))
```
and I think devs would like to have:
```scala
"some".toRight[Throwable]
      .toTry
      .toOption
      .toFuture
```
I was inspired by:
```scala
Try(1).toOption
Try(1).toFuture
Try(1).toEither
```

Comment (taylorleese) [inline common/src/main/scala/com/paypal/cascade/common/option/option.scala]: @michalkowol Try maps to Future much more simply though. Converting an Option to a Future and assuming the None is a failed Future is not intuitive to me. @devmage @2rs2ts Opinions?

Comment (2rs2ts) [inline common/src/main/scala/com/paypal/cascade/common/option/option.scala]: I think this is useful for calling a Java method which you are wrapping in `Option` in a Scala Future flow. It sort of lifts none-ness as a failure into the Future realm. We already had this concept before, now this just provides a default answer for the exception. I think this is an improvement to our current codebase given that observation.

**Comment by devmage** [comment]
@taylorleese I was honestly on the fence about this change, until I came to roughly the same conclusion as @2rs2ts . I don't see myself lifting Options into Future-land with this method all too often, but if it provides value, I'm fine with it.

File: common/src/main/scala/com/paypal/cascade/common/option/option.scala (hunk 1 above)
Comment (devmage) [inline common/src/main/scala/com/paypal/cascade/common/option/option.scala]: @taylorleese I was honestly on the fence about this change, until I came to roughly the same conclusion as @2rs2ts . I don't see myself lifting Options into Future-land with this method all too often, but if it provides value, I'm fine with it.

**Comment by michalkowol** [comment]
I think it will be often used in for-comprehension. That was my use-casse.

Instead:
```scala
for {
  future <- Future(1)
  option <- Option(2).toFuture(new Exception("None"))
} yield {
  future + option
}
```

It will be good to have:
```scala
for {
  future <- Future(1)
  option <- Option(2).toFuture
} yield {
  future + option
}
```

File: common/src/main/scala/com/paypal/cascade/common/option/option.scala (hunk 1 above)
Comment (kionka) [inline common/src/main/scala/com/paypal/cascade/common/option/option.scala]: I really like the ```mapFailure```, but I think this ```toFuture``` is something I don't want to see in production code. The ```Exception``` should need to be explicit in the commons library. This is a little too close to an explicit ```.get``` like ```Future(opt.get)```. Also, this uses method overloading, which I find to be brittle.

Comment (michalkowol) [inline common/src/main/scala/com/paypal/cascade/common/option/option.scala]: I think you are right about `Future(opt.get)`

**Pull request merged**